File: schemaorgphp/classes/service_channel.php
<?php

namespace SchemaOrg;

// ServiceChannel see : https://schema.org/ServiceChannel
class ServiceChannel implements \JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'ServiceChannel';
	
	/**
	 * With properties from Intangible see : https://schema.org/Intangible
	 */
	
	/**
	 * With properties from Thing see : https://schema.org/Thing
	 */
	
	
	/**
	 * An additional type for the item, typically used for adding more specific types from external vocabularies in microdata syntax. This is a relationship between something and a class that the thing is in. In RDFa syntax, it is better to use the native RDFa syntax - the &#39;typeof&#39; attribute - for multiple types. Schema.org tools may have only weaker understanding of extra types, in particular those defined externally.
	 * see : https://schema.org/additionalType
	 * @var string | string[]
	 */
	public $additional_type;
	
	/**
	 * An alias for the item.
	 * see : https://schema.org/alternateName
	 * @var string | string[]
	 */
	public $alternate_name;
	
	/**
	 * A language someone may use with or at the item, service or place. Please use one of the language codes from the IETF BCP 47 standard (see: https://schema.orghttp://tools.ietf.org/html/bcp47). See also inLanguage (see: https://schema.org/inLanguage)
	 * see : https://schema.org/availableLanguage
	 * @var \Language | \Language[] | string | string[]
	 */
	public $available_language;
	
	/**
	 * A description of the item.
	 * see : https://schema.org/description
	 * @var string | string[]
	 */
	public $description;
	
	/**
	 * A sub property of description. A short description of the item used to disambiguate from other, similar items. Information from other properties (in particular, name) may be necessary for the description to be useful for disambiguation.
	 * see : https://schema.org/disambiguatingDescription
	 * @var string | string[]
	 */
	public $disambiguating_description;
	
	/**
	 * The identifier property represents any kind of identifier for any kind of Thing (see: https://schema.org/Thing), such as ISBNs, GTIN codes, UUIDs etc. Schema.org provides dedicated properties for representing many of these, either as textual strings or as URL (URI) links. See background notes (see: https://schema.org/docs/datamodel.html#identifierBg) for more details.
	 * see : https://schema.org/identifier
	 * @var \PropertyValue | \PropertyValue[] | string | string[]
	 */
	public $identifier;
	
	/**
	 * An image of the item. This can be a URL (see: https://schema.org/URL) or a fully described ImageObject (see: https://schema.org/ImageObject).
	 * see : https://schema.org/image
	 * @var \ImageObject | \ImageObject[] | string | string[]
	 */
	public $image;
	
	/**
	 * Indicates a page (or other CreativeWork) for which this thing is the main entity being described. See background notes (see: https://schema.org/docs/datamodel.html#mainEntityBackground) for details. Inverse property: mainEntity (see: https://schema.org/mainEntity).
	 * see : https://schema.org/mainEntityOfPage
	 * @var \CreativeWork | \CreativeWork[] | string | string[]
	 */
	public $main_entity_of_page;
	
	/**
	 * The name of the item.
	 * see : https://schema.org/name
	 * @var string | string[]
	 */
	public $name;
	
	/**
	 * Indicates a potential Action, which describes an idealized action in which this thing would play an &#39;object&#39; role.
	 * see : https://schema.org/potentialAction
	 * @var \Action | \Action[]
	 */
	public $potential_action;
	
	/**
	 * Estimated processing time for the service using this channel.
	 * see : https://schema.org/processingTime
	 * @var \Duration | \Duration[]
	 */
	public $processing_time;
	
	/**
	 * The service provided by this channel.
	 * see : https://schema.org/providesService
	 * @var \Service | \Service[]
	 */
	public $provides_service;
	
	/**
	 * URL of a reference Web page that unambiguously indicates the item&#39;s identity. E.g. the URL of the item&#39;s Wikipedia page, Wikidata entry, or official website.
	 * see : https://schema.org/sameAs
	 * @var string | string[]
	 */
	public $same_as;
	
	/**
	 * The location (e.g. civic structure, local business, etc.) where a person can go to access the service.
	 * see : https://schema.org/serviceLocation
	 * @var \Place | \Place[]
	 */
	public $service_location;
	
	/**
	 * The phone number to use to access the service.
	 * see : https://schema.org/servicePhone
	 * @var \ContactPoint | \ContactPoint[]
	 */
	public $service_phone;
	
	/**
	 * The address for accessing the service by mail.
	 * see : https://schema.org/servicePostalAddress
	 * @var \PostalAddress | \PostalAddress[]
	 */
	public $service_postal_address;
	
	/**
	 * The number to access the service by text message.
	 * see : https://schema.org/serviceSmsNumber
	 * @var \ContactPoint | \ContactPoint[]
	 */
	public $service_sms_number;
	
	/**
	 * The website to access the service.
	 * see : https://schema.org/serviceUrl
	 * @var string | string[]
	 */
	public $service_url;
	
	/**
	 * A CreativeWork or Event about this Thing.. Inverse property: about (see: https://schema.org/about).
	 * see : https://pending.schema.org/subjectOf
	 * @var \CreativeWork | \CreativeWork[] | \Event | \Event[]
	 */
	public $subject_of;
	
	/**
	 * URL of the item.
	 * see : https://schema.org/url
	 * @var string | string[]
	 */
	public $url;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'ServiceChannel'
		);
		
		$serialized = \SchemaOrg\json_serialize( $this->additional_type );
		if ( ! empty( $serialized ) ) {
			$out['additionalType'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->alternate_name );
		if ( ! empty( $serialized ) ) {
			$out['alternateName'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->available_language );
		if ( ! empty( $serialized ) ) {
			$out['availableLanguage'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->description );
		if ( ! empty( $serialized ) ) {
			$out['description'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->disambiguating_description );
		if ( ! empty( $serialized ) ) {
			$out['disambiguatingDescription'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->identifier );
		if ( ! empty( $serialized ) ) {
			$out['identifier'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->image );
		if ( ! empty( $serialized ) ) {
			$out['image'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->main_entity_of_page );
		if ( ! empty( $serialized ) ) {
			$out['mainEntityOfPage'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->name );
		if ( ! empty( $serialized ) ) {
			$out['name'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->potential_action );
		if ( ! empty( $serialized ) ) {
			$out['potentialAction'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->processing_time );
		if ( ! empty( $serialized ) ) {
			$out['processingTime'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->provides_service );
		if ( ! empty( $serialized ) ) {
			$out['providesService'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->same_as );
		if ( ! empty( $serialized ) ) {
			$out['sameAs'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->service_location );
		if ( ! empty( $serialized ) ) {
			$out['serviceLocation'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->service_phone );
		if ( ! empty( $serialized ) ) {
			$out['servicePhone'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->service_postal_address );
		if ( ! empty( $serialized ) ) {
			$out['servicePostalAddress'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->service_sms_number );
		if ( ! empty( $serialized ) ) {
			$out['serviceSmsNumber'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->service_url );
		if ( ! empty( $serialized ) ) {
			$out['serviceUrl'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->subject_of );
		if ( ! empty( $serialized ) ) {
			$out['subjectOf'] = $serialized;
		}
		
		$serialized = \SchemaOrg\json_serialize( $this->url );
		if ( ! empty( $serialized ) ) {
			$out['url'] = $serialized;
		}
		
		return $out;
	}
}

File: releases/v3.1.0/schemaorggo/service_channel.go
package schemaorggo

import "encoding/json"

// ServiceChannel see : https://schema.org/ServiceChannel
type ServiceChannel struct {

	// With properties from Intangible see : https://schema.org/Intangible
	//

	// With properties from Thing see : https://schema.org/Thing
	//

	// AdditionalType see : https://schema.org/additionalType
	// An additional type for the item, typically used for adding more specific types from external vocabularies in microdata syntax. This is a relationship between something and a class that the thing is in. In RDFa syntax, it is better to use the native RDFa syntax - the &#39;typeof&#39; attribute - for multiple types. Schema.org tools may have only weaker understanding of extra types, in particular those defined externally.
	// types : URL
	AdditionalType []string `json:"additionalType,omitempty"`

	// AlternateName see : https://schema.org/alternateName
	// An alias for the item.
	// types : Text
	AlternateName []string `json:"alternateName,omitempty"`

	// AvailableLanguage see : https://schema.org/availableLanguage
	// A language someone may use with the item. Please use one of the language codes from the [IETF BCP 47 standard](http://tools.ietf.org/html/bcp47). See also [[inLanguage]]
	// types : Language Text
	AvailableLanguage []interface{} `json:"availableLanguage,omitempty"`

	// Description see : https://schema.org/description
	// A description of the item.
	// types : Text
	Description []string `json:"description,omitempty"`

	// DisambiguatingDescription see : https://schema.org/disambiguatingDescription
	// A sub property of description. A short description of the item used to disambiguate from other, similar items. Information from other properties (in particular, name) may be necessary for the description to be useful for disambiguation.
	// types : Text
	DisambiguatingDescription []string `json:"disambiguatingDescription,omitempty"`

	// Image see : https://schema.org/image
	// An image of the item. This can be a [[URL]] or a fully described [[ImageObject]].
	// types : URL ImageObject
	Image []interface{} `json:"image,omitempty"`

	// MainEntityOfPage see : https://schema.org/mainEntityOfPage
	// Indicates a page (or other CreativeWork) for which this thing is the main entity being described. See [background notes](/docs/datamodel.html#mainEntityBackground) for details.
	// types : CreativeWork URL
	MainEntityOfPage []interface{} `json:"mainEntityOfPage,omitempty"`

	// Name see : https://schema.org/name
	// The name of the item.
	// types : Text
	Name []string `json:"name,omitempty"`

	// PotentialAction see : https://schema.org/potentialAction
	// Indicates a potential Action, which describes an idealized action in which this thing would play an &#39;object&#39; role.
	// types : Action
	PotentialAction []*Action `json:"potentialAction,omitempty"`

	// ProcessingTime see : https://schema.org/processingTime
	// Estimated processing time for the service using this channel.
	// types : Duration
	ProcessingTime []*Duration `json:"processingTime,omitempty"`

	// ProvidesService see : https://schema.org/providesService
	// The service provided by this channel.
	// types : Service
	ProvidesService []*Service `json:"providesService,omitempty"`

	// SameAs see : https://schema.org/sameAs
	// URL of a reference Web page that unambiguously indicates the item&#39;s identity. E.g. the URL of the item&#39;s Wikipedia page, Freebase page, or official website.
	// types : URL
	SameAs []string `json:"sameAs,omitempty"`

	// ServiceLocation see : https://schema.org/serviceLocation
	// The location (e.g. civic structure, local business, etc.) where a person can go to access the service.
	// types : Place
	ServiceLocation []*Place `json:"serviceLocation,omitempty"`

	// ServicePhone see : https://schema.org/servicePhone
	// The phone number to use to access the service.
	// types : ContactPoint
	ServicePhone []*ContactPoint `json:"servicePhone,omitempty"`

	// ServicePostalAddress see : https://schema.org/servicePostalAddress
	// The address for accessing the service by mail.
	// types : PostalAddress
	ServicePostalAddress []*PostalAddress `json:"servicePostalAddress,omitempty"`

	// ServiceSmsNumber see : https://schema.org/serviceSmsNumber
	// The number to access the service by text message.
	// types : ContactPoint
	ServiceSmsNumber []*ContactPoint `json:"serviceSmsNumber,omitempty"`

	// ServiceUrl see : https://schema.org/serviceUrl
	// The website to access the service.
	// types : URL
	ServiceUrl []string `json:"serviceUrl,omitempty"`

	// Url see : https://schema.org/url
	// URL of the item.
	// types : URL
	Url []string `json:"url,omitempty"`
}

func (v ServiceChannel) MarshalJSON() ([]byte, error) {
	type Alias ServiceChannel

	b, err := json.Marshal((Alias)(v))
	if err != nil {
		return nil, err
	}

	return append([]byte("{\"@context\":\"http://schema.org\",\"@type\":\"ServiceChannel\","), b[1:]...), nil
}
